lib/connection/pool.js

```diff
@@ -581,6 +581,13 @@ function messageHandler(self) {
           }
 
           if (responseDoc.writeConcernError) {
+            if (responseDoc.ok === 1) {
```

Comment:
What about doing a ternary to create the error, and then using a single callback?

```js
if (responseDoc.writeConcernError) {
  const err = responseDoc.ok === 1 ?
    new MongoWriteConcernError(responseDoc.writeConcernError, responseDoc) :
    new MongoWriteConcernError(responseDoc.writeConcernError);
  return handleOperationCallback(self, workItem.cb, err);
}
```